###LOAD DATA

---



In [2]:
import pandas as pd

data = pd.read_csv("/content/transactions.csv");
data

transaction_id  product_id  customer_id transaction_date online_order  \
0                   1           2         2950        25/2/2017        False   
1                   2           3         3120        21/5/2017         True   
2                   3          37          402       16/10/2017        False   
3                   4          88         3135        31/8/2017        False   
4                   5          78          787        1/10/2017         True   
...               ...         ...          ...              ...          ...   
19995           19996          51         1018        24/6/2017         True   
19996           19997          41          127        9/11/2017         True   
19997           19998          87         2284        14/4/2017         True   
19998           19999           6         2764         3/7/2017        False   
19999           20000          11         1144        22/9/2017         True   

      order_status           brand product_line product_class product_size  \
0         Approved           Solex     Standard        medium       medium   
1         Approved   Trek Bicycles     Standard        medium        large   
2         Approved      OHM Cycles     Standard           low       medium   
3         Approved  Norco Bicycles     Standard        medium       medium   
4         Approved  Giant Bicycles     Standard        medium        large   
...            ...             ...          ...           ...          ...   
19995     Approved      OHM Cycles     Standard          high       medium   
19996     Approved           Solex         Road        medium       medium   
19997     Approved      OHM Cycles     Standard        medium       medium   
19998     Approved      OHM Cycles     Standard          high       medium   
19999     Approved   Trek Bicycles     Standard        medium        small   

       list_price standard_cost  product_first_sold_date  
0           71.49        $53.62                  41245.0  
1         2091.47       $388.92                  41701.0  
2         1793.43       $248.82                  36361.0  
3         1198.46       $381.10                  36145.0  
4         1765.30       $709.48                  42226.0  
...           ...           ...                      ...  
19995     2005.66     $1,203.40                  37823.0  
19996      416.98       $312.74                  35560.0  
19997     1636.90        $44.71                  40410.0  
19998      227.88       $136.73                  38216.0  
19999     1775.81     $1,580.47                  36334.0  

[20000 rows x 13 columns]

###DATA INFORMATION

---



In [3]:
print("DATA DESCRIPTION")
print(data.describe())
print()

print("DATATYPES")
print(data.info())
print()

print("COLUMNS")
print(data.columns)
print()

DATA DESCRIPTION
       transaction_id   product_id   customer_id    list_price  \
count    20000.000000  20000.00000  20000.000000  20000.000000   
mean     10000.500000     45.36465   1738.246050   1107.829449   
std       5773.647028     30.75359   1011.951046    582.825242   
min          1.000000      0.00000      1.000000     12.010000   
25%       5000.750000     18.00000    857.750000    575.270000   
50%      10000.500000     44.00000   1736.000000   1163.890000   
75%      15000.250000     72.00000   2613.000000   1635.300000   
max      20000.000000    100.00000   5034.000000   2091.470000   

       product_first_sold_date  
count             19803.000000  
mean              38199.776549  
std                2875.201110  
min               33259.000000  
25%               35667.000000  
50%               38216.000000  
75%               40672.000000  
max               42710.000000  

DATATYPES
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20000 entries, 0 to 19999
Data

###HANDLING NULL VALUES

---



####Checking for null values

In [4]:
data.isnull().sum()

transaction_id               0
product_id                   0
customer_id                  0
transaction_date             0
online_order               360
order_status                 0
brand                      197
product_line               197
product_class              197
product_size               197
list_price                   0
standard_cost              197
product_first_sold_date    197
dtype: int64

####Dropna on **'brand'** column

Dropping the null values in ***'brand***' column as the total null values in that column are **197** which is significantly smaller as compared to total rows in the dataset i.e **20,000**

By dropping the null values in ***brand*** column null values in the follwing columns are also dropped:
***product_line, product_class, product_size, standard_cost, product_first_sold_date***

In [5]:
data.dropna(subset=['brand'], inplace=True)
data.isnull().sum()

transaction_id               0
product_id                   0
customer_id                  0
transaction_date             0
online_order               358
order_status                 0
brand                        0
product_line                 0
product_class                0
product_size                 0
list_price                   0
standard_cost                0
product_first_sold_date      0
dtype: int64

####***'online_order'*** column

Filling the null values in ***online_order*** column with mode value that column i.e ***'True'*** since deleting the columns can make are dataset short because their are total of *358* null values in online_order column.

In [13]:
print("Mode = ",data['online_order'].mode())
print((data['online_order']==True).sum(), "occurrence")
print()

data['online_order'] = data['online_order'].fillna("True");
data.isnull().sum()

Mode =  0    True
Name: online_order, dtype: object
9739 occurrence



transaction_id             0
product_id                 0
customer_id                0
transaction_date           0
online_order               0
order_status               0
brand                      0
product_line               0
product_class              0
product_size               0
list_price                 0
standard_cost              0
product_first_sold_date    0
dtype: int64

###OUTLIERS

---

No outliers found in the dataset

In [15]:
value_column = 'list_price'
outliers = data[(data[value_column] < data[value_column].mean() - 2 * data[value_column].std()) |
              (data[value_column] > data[value_column].mean() + 2 * data[value_column].std())]

print("Outliers:\n")
outliers

Outliers:



Empty DataFrame
Columns: [transaction_id, product_id, customer_id, transaction_date, online_order, order_status, brand, product_line, product_class, product_size, list_price, standard_cost, product_first_sold_date]
Index: []

###Saving the Cleaned Data

---



In [16]:
data.to_csv(r"cleaned_data2.csv");